server/routes/prediction.routes.js

```diff
@@ -359,146 +364,155 @@ function normalizeMatchFilter(filter, field){
-        if (configuration.getConfig("minPredictionCutoffDate")) {
-          attributes.where.date = {[Op.gt]: configuration.getConfig("minPredictionCutoffDate")}
-        } else if (configuration.getConfig("predictionCutoffDays")) {
-          const numDays = configuration.getConfig("predictionCutoffDays")
-          const today = new Date()
-          let cutoff = new Date()
-          cutoff.setDate(today.getDate() - numDays)
-          attributes.where.date = {[Op.gt]: cutoff}
+    // Process regular filters
+    if (filter.filters) {
+      Object.entries(filter.filters).forEach(([field, data]) => {
+        if (data.matchMode === 'equals' && data.value) {
+          attributes.where[field] = data.value; // Simplified without Op.eq
         }
+      });
+      logger.debug('Attributes after applying filters', { attributes });
+    }
+
+    // Handle date filtering
+    if (!filter.ignoreDateCutoff && !filter.filters?.solNum) {
+      if (configuration.getConfig('minPredictionCutoffDate')) {
+        attributes.where.date = { [Sequelize.Op.gt]: configuration.getConfig('minPredictionCutoffDate') };
+      } else if (configuration.getConfig('predictionCutoffDays')) {
+        const cutoff = new Date();
+        cutoff.setDate(cutoff.getDate() - configuration.getConfig('predictionCutoffDays'));
+        attributes.where.date = { [Sequelize.Op.gt]: cutoff };
       }
     }
 
-
-
     if (filter.startDate) {
-      // double check they aren't asking for data from before the cutoff
-      const start = Date.parse(filter.startDate)
-      const cutoff = Date.parse(configuration.getConfig("minPredictionCutoffDate", '1990-01-01'))
+      const start = Date.parse(filter.startDate);
+      const cutoff = Date.parse(configuration.getConfig('minPredictionCutoffDate', '1990-01-01'));
       if (start > cutoff) {
-        attributes.where.date = { [Op.gt]: filter.startDate }
+        attributes.where.date = {
+          ...attributes.where.date,
+          [Sequelize.Op.gt]: filter.startDate,
+        };
       }
     }
 
     if (filter.endDate) {
-      attributes.where.date = (attributes.where.date) ?
-        Object.assign(attributes.where.date, { [Op.lt]: filter.endDate }) :
-        { [Op.lt]: filter.endDate }
+      attributes.where.date = {
+        ...attributes.where.date,
+        [Sequelize.Op.lt]: filter.endDate,
+      };
     }
 
-    // finally, put in an agency filter if this user isn't an admin
-    // want to do it last so it overrides any possible agency setting in the supplied filter
-    if ( ! authRoutes.isGSAAdmin(user.agency, user.userRole)) {
-      attributes.where.agency  = {
-        [Op.eq] : (user && user.agency) ? user.agency : ''
-      }
-    }
+    // Agency/Office filtering for non-admin users
+    if (!isGSAAdmin(user.agency, user.userRole)) {
+      // Apply agency and computed office filters directly
+      attributes.where.agency = user.agency;
+      attributes.where.office = user.office;
```

Comment:
How are you going to pull RFQ's that have a agency value that would normally be an office value?

For instance, Department of the Navy

Reply:
The `applyAgencyOfficeFilters` function now handles this case by checking both fields:
```javascript
if (officeFilter) {
  attributes.where[Op.or] = [
    { office: officeFilter },
    { agency: officeFilter }  // This catches cases where EBUY data puts office in agency field
  ];
}
```